app/code/Magento/GoogleGtag/view/frontend/templates/ga.phtml

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+?>
+<?php /** @var $block \Magento\GoogleGtag\Block\Ga */ ?>
+<?php /** @var \Magento\GoogleGtag\Helper\GtagConfiguration $gtagConfiguration */ ?>
+<?php  $gtagConfiguration = $block->getHelper() ?>
+<?php  $measurementId = $gtagConfiguration->getMeasurementId() ?>
+<!-- BEGIN GOOGLE ANALYTICS 4 CODE -->
+<script type="text/x-magento-init">
+{
+    "*": {
+        "Magento_GoogleGtag/js/google-analytics": {
+            "isCookieRestrictionModeEnabled": <?= (int)$block->isCookieRestrictionModeEnabled() ?>,
+            "currentWebsite": <?= (int)$block->getCurrentWebsiteId() ?>,
+            "cookieName": "<?= /* @noEscape */ \Magento\Cookie\Helper\Cookie::IS_USER_ALLOWED_SAVE_COOKIE ?>",
+            "pageTrackingData": <?= /* @noEscape */ json_encode($block->getPageTrackingData($measurementId)) ?>,
```

Comment:
I would suggest serializing it in the block (or view model) using an api like `\Magento\Framework\Serialize\SerializerInterface`